src/server.ml

```diff
@@ -272,8 +296,10 @@ let rec wait_loop process_params verbose accept =
 					) m.m_types;
 					if m.m_extra.m_kind <> MSub then Typeload.add_module ctx m p;
 					PMap.iter (Hashtbl.replace com2.resources) m.m_extra.m_binded_res;
-					PMap.iter (fun _ m2 -> add_modules (tabs ^ "  ") m0 m2) m.m_extra.m_deps);
+					if ctx.Typecore.in_macro || com2.display.dms_full_typing then
```

Comment:
@ncannasse: I stared at this particular change some more and realized that it would potentially miss dealing with `m_binded_res` and `m_macro_calls` for dependencies. I'm pretty sure we don't care about resources in display mode, but I have no idea what `m_macro_calls` is used for.

What exactly do you do with `Context.registerModuleReuseCall`, and is it something that's important for display mode?